lib/websocket.js

```diff
@@ -1188,7 +1194,10 @@ function receiverOnError(err) {
     websocket.close(err[kStatusCode]);
   }
 
-  websocket.emit('error', err);
+  if (!websocket._errorEmitted) {
+    websocket._errorEmitted = true;
+    websocket.emit('error', err);
```

Comment:
This seems worthwhile to pull out into a small function:

```js
function emitError(err) {
  websocket._errorEmitted = true;
  websocket.emit('error', err);
}
```

Seems like these lines should always be called together. Pulling them out into a function will make it less likely that someone will forget to do so in the future.